### DISCUSSION TITLE
Use libray as IR barrier

### DISCUSSION THREAD

**Discussion by Manhem** [bengtmartensson/Infrared4Arduino#75] [General]
Hi.
Is it possible to use your library as a IR barrier.
Send and receive at the same time from the same processor?
And can I also use a carrier 38KHz and modulate it with for example 1KHz?
And if so. Do you have an example to share?

**Comment by bengtmartensson** [comment]
Hi, 

> Is it possible to use your library as a IR barrier.

most often, the term "IR translator" is used; you will get more Google hits with that word... Or possibly you mean something else?

> Send and receive at the same time from the same processor?

Assuming you definition of "at the same time" includes sending immediately after reception, yes, this can be done. An advantage of using one processor operating sequentially is that the receiver can be turned off while sending, eliminating spill-overs from the sender to the receiver.

> And can I also use a carrier 38KHz and modulate it with for example 1KHz?

Sure.
 
> Do you have an example to share?

Not yet, but it is really quite straightforward. Feel free to submit one...

**Comment by Manhem** [comment]
Hi.
The things I like to do is sending one modulated signal to an reflector and then receive it on the way back.
If the beam is broken, I would like to have a signal so I can get an alarm.
I shall use this to see if someone passing the road to my house.
If this is possible, can you write a example so I can see how your library works.

**Reply by bengtmartensson** [reply]
Ok, now your term "IR barrier" makes sense.  That is not the use case that my library deals with ([consumer IR](https://en.wikipedia.org/wiki/Consumer_IR)).

Check out the [Arduino forum, subforum sensors](https://forum.arduino.cc/c/using-arduino/sensors/26); I think I have seen some useful stuff there.

**Comment by Manhem** [comment]
Thanks.
I try that.
